lib/resty/etcd/v3.lua

```diff
@@ -196,6 +272,17 @@ local function choose_endpoint(self)
         return endpoints[1]
     end
 
+    -- if enable health check
+    if self.max_fails then
+        for _, endpoint in ipairs(endpoints) do
+            if endpoint.health_status == 1 then
+                return endpoint
+            end
+        end
+        utils.log_error("has no health etcd endpoint")
+        return nil
```

Comment:
As we don't check if `choose_endpoint` returns `nil`, this change will cause an error to throw. It is bad to use throwing error as a control flow. Although APISIX captures the error, other users may not do this.

We should return `nil, err` here and check it outside of `choose_endpoint`.